gslib/tests/test_stat.py

```diff
@@ -141,9 +141,9 @@ def test_stat_one_missing_wildcard(self):
   def test_stat_bucket_wildcard(self):
     bucket_uri = self.CreateBucket()
     self.CreateObject(bucket_uri=bucket_uri, object_name='foo', contents=b'z')
-    stat_string = suri(bucket_uri)[:-1] + '?/foo'
+    stat_string = suri(bucket_uri) + '/?oo'
```

Comment:
This test is called `test_stat_bucket_wildcard`, but you're removing the wildcard from the bucket portion of the URI :P

It's interesting that this fails when more than 1k buckets are present in the project -- the prefix we give before the wildcard is *the full bucket name*, so I'd assume that the listing call would use the bucket name a prefix for the listing call-- oooh, wait. That's right. The XML API doesn't have the `prefix` field for bucket listing requests:
https://cloud.google.com/storage/docs/xml-api/get-service

...like the JSON API does:
https://cloud.google.com/storage/docs/json_api/v1/buckets/list

Yay for interoperability with other poorly-designed storage providers' APIs :expressionless: 

...all that being said, we're stuck with this shortcoming of the XML API... but I can think of one disgusting hack to get around it.  The bucket listing is ordered lexicographically, so you're going to get the first 1000 buckets in nicely sorted order.  So for the buckets we use for these tests, we could just start their names with an alphabetically lesser value... I think `CreateBucket()` will create you a name starting with `gsutil-test-` if you don't supply it with one. So, let's prefix these buckets' names with something like `aaa-list-sort-`.  Should be doable via:

```
bucket_name = self.MakeTempname('bucket', prefix='aaa-list-sort-')
self.CreateBucket(bucket_name)
```

This applies here and in the other tests.

Reply:
Thanks for all the context, and whoops, my mistake swapping the wildcard to the object! I'll edit up this PR. :)

Reply:
Ok, applied prefixes. Let me know if naming is ok, and if this looks like an ok spot for the new function to live. 👍